### PR TITLE
Probabilistic Forecast timeseries in plotting module

### LINKED ISSUE #550: Add ProbabilisticForecastGroup plotting to plotting module
Related to https://github.com/SolarArbiter/solarforecastarbiter-dashboard/issues/325, Add a `generate_probabilistic_forecast_figure` function to the `solarforecastarbiter.plotting.timeseries` module. Should display the same plotly plot we use in reports. Documentation should make sure to point out that the function returns a plotly figure, and not bokeh.

### PR DESCRIPTION
**Pull request by lboeman**
- [x] Closes #550  .
  - [x] I am familiar with the [contributing guidelines](https://solarforecastarbiter-core.readthedocs.io/en/latest/contributing.html).
  - [x] Tests added.
  - [x] Updates entries to [`docs/source/api.rst`](https://github.com/SolarArbiter/solarforecastarbiter-core/blob/master/docs/source/api.rst) for API changes.
  - [x] Adds descriptions to appropriate "what's new" file in [`docs/source/whatsnew`](https://github.com/SolarArbiter/solarforecastarbiter-core/tree/master/docs/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
  - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
  - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

**Comment by lboeman** [comment]
Adds probabilistic forecast plotting to the `solarforecastarbiter.plotting.timeseries` module. This largely copies the logic from `reports.figures.plotly_figures`. There is a good amount of copied code with slight alteration, but it seemed as though adding code to coerce to or from the metadata and value dataframes in `reports.figures.plotly_figures` would make the code harder to understand, so I didn't spend too much time trying to factor out common functions. I believe the initial purpose of these dataframes were to reduce the size of the Bokeh ColumnDataSource objects, and they may no longer be needed. 

Screenshots of the plots produced here can be found in the corresponding dashboard PR: https://github.com/SolarArbiter/solarforecastarbiter-dashboard/pull/351

**Review by wholmgren — APPROVED** [review]
Looks good to me. Maybe better to address copy pasta in a bigger plotting clean up. What do you think @alorenzo175?

File: solarforecastarbiter/plotting/timeseries.py [hunk 1]
```diff
@@ -336,3 +339,169 @@ def generate_observation_figure(observation, data, limit=pd.Timedelta('3d')):
...
+    logger.info('Starting probabilistic forecast figure generation...')
+    if len(data.index) == 0:
+        return None
+
+    fig = go.Figure()
+    if 'x' in forecast.axis:
+        ylabel = 'Probability (%)'
+        _plot_probabilsitic_distribution_axis_x(fig, forecast, data)
+    else:
+        ylabel = f'Data ({forecast.units})'
```
Comment (alorenzo175) [inline solarforecastarbiter/plotting/timeseries.py:491]:
```suggestion
        ylabel = plot_utils.format_variable_name(forecast.variable)
```

**Review by alorenzo175 — COMMENTED** [review]
I agree that we can wait on the copy pasta for when the bokeh plots are replaced w/ plotly

**Comment by lboeman** [comment]
I'm not sure what's up with the codecov report, it's saying that tests are not covered. Other than that If you guys don't have any other concerns I think this is good to go?

**Review by alorenzo175 — APPROVED** [review]
good with me

**Pull request merged**